tests/std/tests/P0896R4_views_common/test.cpp

```diff
@@ -0,0 +1,383 @@
+
+        // ... with const lvalue argument
+        static_assert(CanViewCommon<const remove_reference_t<Rng>&> == (!is_view || copyable<V>) );
+        if constexpr (is_view && copyable<V>) {
+            constexpr bool is_noexcept = is_nothrow_copy_constructible_v<V>;
+
+            static_assert(same_as<decltype(views::common(as_const(rng))), R>);
+            static_assert(noexcept(views::common(as_const(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | views::common), R>);
+            static_assert(noexcept(as_const(rng) | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | views::common | views::common | views::common), V>);
+            static_assert(noexcept(as_const(rng) | views::common | views::common | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | pipeline), V>);
+            static_assert(noexcept(as_const(rng) | pipeline) == is_noexcept);
+        } else if constexpr (!is_view) {
+            using RC                   = common_view<views::all_t<const remove_reference_t<Rng>&>>;
+            constexpr bool is_noexcept = is_nothrow_constructible_v<RC, const remove_reference_t<Rng>&>;
+
+            static_assert(same_as<decltype(views::common(as_const(rng))), RC>);
+            static_assert(noexcept(views::common(as_const(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | views::common), RC>);
+            static_assert(noexcept(as_const(rng) | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | views::common | views::common | views::common), RC>);
+            static_assert(noexcept(as_const(rng) | views::common | views::common | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(as_const(rng) | pipeline), RC>);
+            static_assert(noexcept(as_const(rng) | pipeline) == is_noexcept);
+        }
+
+        // ... with rvalue argument
+        static_assert(CanViewCommon<remove_reference_t<Rng>> == is_view || enable_borrowed_range<remove_cvref_t<Rng>>);
+        if constexpr (is_view) {
+            constexpr bool is_noexcept = is_nothrow_move_constructible_v<V>;
+            static_assert(same_as<decltype(views::common(move(rng))), R>);
+            static_assert(noexcept(views::common(move(rng))) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common), R>);
+            static_assert(noexcept(move(rng) | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | views::common | views::common | views::common), R>);
+            static_assert(noexcept(move(rng) | views::common | views::common | views::common) == is_noexcept);
+
+            static_assert(same_as<decltype(move(rng) | pipeline), R>);
+            static_assert(noexcept(move(rng) | pipeline) == is_noexcept);
+        } else if constexpr (enable_borrowed_range<Rng>) {

```

Comment:
I observe variation in the template argument given to `enable_borrowed_range` in the `static_assert` above, followed by the `if constexpr` here. This occurs (somewhat differently) elsewhere in the test. @CaseyCarter recalls that there's a reason for this, but not the exact reason. A comment might be nice.

Reply:
Hm I cannot really find a reason for this. @CaseyCarter ?

Reply:
There was something important I needed to remember here: the fact that we always `remove_cvref_t` the argument to `enable_borrowed_range`. The library doesn't even allow user specializations for cv-qualified types or reference types. So this should be:
```suggestion
        } else if constexpr (enable_borrowed_range<remove_cvref_t<Rng>>) {
```
and I should audit for occurrences in other tests.

Reply:
I pushed a commit to fix four occurrences in this test.